### DISCUSSION TITLE
Support Deutsche Kredit Bank (DKB) balances and transactions query

### DISCUSSION THREAD

**Discussion by LefterisJP** [rotki/rotki#6699] [Ideas]
## Abstract

As a local application we can utilize the FINTS protocol that almost all German banks are implementing and offer the ability to automatically fetch user balances, statements and holdings:

https://github.com/raphaelm/python-fints.

## Task

Use https://github.com/raphaelm/python-fints and experiment with how a generic user interface should look like. Experiment with N26, DKB and ING-DiBa which our team has access to and see if they work fine.

**Comment by LefterisJP** [comment]
We could also use plaid or open banking integration for Europe. An example app that does this: https://github.com/kevinschaich/mintable

**Comment by isidorosp** [comment]
check also https://www.saltedge.com/products/psd2_account_info and https://www.yodlee.com/europe/account-verification

**Comment by step21** [comment]
Here is another example, where a (proprietary) app uses a small service that has the PSD2 license, to access banks that use it. https://moneymoney-app.com/psd2/
For FinTS this is not required, but for everything that uses PSD2 iirc.

**Comment by LefterisJP** [comment]
I think we need work per bank. Either find libraries that do it, or even write scrapers ourselves. It's doable.

So I will turn this issue in order to support my own bank. DKB and then we can slowly add support for more banks.

**Comment by LefterisJP** [comment]
In general we can see if we could also use OFX https://financialdataexchange.org/ofx

**Comment by step21** [comment]
There are also similar services that are not ‘open’ but then again also FDX costs money. Example: https://nordigen.com/en (And since shortly banks have to offer access via PSD2, but it builds on mediated access via companies, not directly by consumer)
One app I use for macos, money money has a subscription (cheap) only to access PSD2 accounts or others that cannot be queried locally. They are also a Berlin based company if you want to ask them how they did that.

**Comment by LefterisJP** [comment]
> There are also similar services that are not ‘open’ but then again also FDX costs money. Example: https://nordigen.com/en (And since shortly banks have to offer access via PSD2, but it builds on mediated access via companies, not directly by consumer) One app I use for macos, money money has a subscription (cheap) only to access PSD2 accounts or others that cannot be queried locally. They are also a Berlin based company if you want to ask them how they did that.

We may use such services in the future, but we would need to ask users to register and pay.

Another approach many local apps use and one which I will start with is to simply authenticate with the bank for a small session and pull data. And each time you need to authenticate just prompt for a 2FA from the user. This is how I managed to do DKB. Many scripts exist online for various banks for this. Ours is gonna be based on the work done by these guys: https://github.com/grindsa/dkb-robo

Just will need to port their code as we can't use it as is inside rotki.

**Comment by LefterisJP** [comment]
Related discussion: https://github.com/rotki/rotki/discussions/4640

**Comment by LefterisJP** [comment]
Wish we could somehow merge [#4640](https://github.com/rotki/rotki/discussions/4640) and this
